hyperspy/axes.py

```diff
@@ -825,6 +990,98 @@ def _on_scale_changed(self):
+            `navigation` or `signal`. By default the converted units of the 
+            first axis is used for all axes. If `False`, convert all axes 
+            individually.
+        factor : float
+            Factor used to determine the prefix of the units.
+        """
+        convert_navigation = convert_signal = True
+
+        if axes is None:
+            axes = self.navigation_axes + self.signal_axes
+        elif axes == 'navigation':
+            axes = self.navigation_axes
+            convert_signal = False
+        elif axes == 'signal':
+            axes = self.signal_axes
+            convert_navigation = False
+        elif isinstance(axes, DataAxis) or type(axes) is int or type(axes) is str:
+            if not isinstance(axes, DataAxis):
+                axes = self[axes]
+            axes = (axes, )
+            convert_navigation = axes[0].navigate
+            convert_signal = not axes[0].navigate
+        else:
+            raise TypeError(
+                'Axes type `{}` is not correct.'.format(type(axes)))
+
+        if type(units) is str or units is None:
+            units = [units] * len(axes)
+        elif type(units) is list:
+            if len(units) != len(axes):
+                raise ValueError('Length of the provided units list {} should '
+                                 'be the same than the length of the provided '
+                                 'axes {}.'.format(units, axes))
+        else:
+            raise TypeError('Units type `{}` is not correct. It can be a '
+                            '`string`, a `list` of string or `None`.'
+                            ''.format(type(units)))
+
+        if same_units:
+            if convert_navigation:
+                units_nav = units[:self.navigation_dimension]
+                self._convert_axes_to_same_units(self.navigation_axes,
+                                                 units_nav, factor)
+            if convert_signal:
+                offset = self.navigation_dimension if convert_navigation else 0
+                units_sig = units[offset:]
+                self._convert_axes_to_same_units(self.signal_axes,
+                                                 units_sig, factor)
+        else:
+            for axis, unit in zip(axes, units):
+                axis.convert_to_units(unit, factor=factor)
+
+    def _convert_axes_to_same_units(self, axes, units, factor=0.25):
+        current_units = None
+        for axis, unit in zip(axes, units):
+            axis.convert_to_units(unit, factor)
+            if current_units is not None and current_units:
+                if current_units != axis.units:
+                    # take units along first axis and use that one
+                    for axis in axes:
+                        axis.convert_to_units(axes[0].units, factor)
+                    return
+            else:
+                current_units = axis.units  # initialisation first iteration
```

Comment:
If you only intend to keep the first `is not None and len(unit)` unit, why not find it first from the list, and then loop over the axes just once? 
If this seems acceptable, how about this:
```python
first_good_unit = next(filter(lambda x: x is not None and x, units))
# Change `micron` to `um` if you want here as well
for axis in filter(lambda ax: ax.unit != first_good_unit, axes):
    axis.convert_to_units(first_good_unit, factor)
```

Reply:
I simplified here, actually it was very simple and really unnecessary complicated!!